time/src/main/java/net/jqwik/time/internal/properties/arbitraries/DefaultLocalDateTimeArbitrary.java

```diff
@@ -0,0 +1,151 @@
+			return times[1];
+		}
+	}
+
+	private LocalDateTime calculateEffectiveMin() {
+		LocalDateTime effective = min != null ? min : DEFAULT_MIN;
+		return calculateEffectiveMinWithPrecision(effective);
+	}
+
+	private LocalDateTime calculateEffectiveMinWithPrecision(LocalDateTime effective) {
+		LocalDate date = effective.toLocalDate();
+		LocalTime time = effective.toLocalTime();
+		try {
+			time = DefaultLocalTimeArbitrary.calculateEffectiveMinWithPrecision(time, ofPrecision);
+		} catch (IllegalArgumentException e) {
+			time = LocalTime.MIN;
+			LocalDate effectiveDate;
+			try {
+				effectiveDate = date.plusDays(1);
+			} catch (DateTimeException dateTimeException) {
+				throw new IllegalArgumentException("Date is LocalDate.MAX and must be increased by 1 day.");
+			}
+			date = effectiveDate;
+		}
+		return LocalDateTime.of(date, time);
+	}
+
+	private LocalDateTime calculateEffectiveMax() {
+		LocalDateTime effective = max != null ? max : DEFAULT_MAX;
+		return calculateEffectiveMaxWithPrecision(effective);
+	}
+
+	private LocalDateTime calculateEffectiveMaxWithPrecision(LocalDateTime effective) {
+		LocalDate date = effective.toLocalDate();
+		LocalTime time = effective.toLocalTime();
+		time = DefaultLocalTimeArbitrary.calculateEffectiveMaxWithPrecision(time, ofPrecision);
+		return LocalDateTime.of(date, time);
+	}
+
+	private LocalTimeArbitrary[] generateTimeArbitrary(LocalDateTime effectiveMin, LocalDateTime effectiveMax, ChronoUnit ofPrecision) {
+		LocalTimeArbitrary[] times = new LocalTimeArbitrary[3];
+		if (effectiveMin.toLocalDate().isEqual(effectiveMax.toLocalDate())) {
+			times[0] = Times.times().between(effectiveMin.toLocalTime(), effectiveMax.toLocalTime()).ofPrecision(ofPrecision);
+		} else {
+			times[0] = Times.times().atTheEarliest(effectiveMin.toLocalTime()).ofPrecision(ofPrecision);
+			times[1] = Times.times().ofPrecision(ofPrecision);
+			times[2] = Times.times().atTheLatest(effectiveMax.toLocalTime()).ofPrecision(ofPrecision);
+		}
+		return times;
+	}
+
+	@Override
+	public LocalDateTimeArbitrary atTheEarliest(LocalDateTime min) {
+		if (min.getYear() <= 0) {
+			throw new IllegalArgumentException("Minimum year in a date time must be > 0");
+		}
+		if ((max != null) && min.isAfter(max)) {
+			throw new IllegalArgumentException("Minimum date time must not be after maximum date time");
+		}
+
+		DefaultLocalDateTimeArbitrary clone = typedClone();
+		clone.min = min;
+		return clone;
+	}
+
+	@Override
+	public LocalDateTimeArbitrary atTheLatest(LocalDateTime max) {
+		if (max.getYear() <= 0) {
+			throw new IllegalArgumentException("Minimum year in a date time must be > 0");
+		}
+		if ((min != null) && max.isBefore(min)) {
+			throw new IllegalArgumentException("Maximum date time must not be before minimum date time");
+		}
+
+		DefaultLocalDateTimeArbitrary clone = typedClone();
+		clone.max = max;
+		return clone;
+	}
+
+	@Override
+	public LocalDateTimeArbitrary ofPrecision(ChronoUnit ofPrecision) {
```

Comment:
Maybe introduce constant set of allowed precisions and check here for inclusion.